### PR TITLE
CB-6127 Improve FreeIPA service termination trigger

### PR DESCRIPTION
**Pull request by lacikaaa**

### CODE REVIEW

File: datalake/src/test/java/com/sequenceiq/datalake/service/sdx/SdxServiceTest.java [hunk 1]
```diff
@@ -124,6 +125,9 @@
     @Mock
     private CDPConfigService cdpConfigService;
 
+    @Mock
+    private FlowCancelService flowCancelServicel;
```
Comment (topolyai5) [inline datalake/src/test/java/com/sequenceiq/datalake/service/sdx/SdxServiceTest.java]: typo in `flowCancelServicel` -> `flowCancelService`

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/flow/FreeIpaFlowManager.java [hunk 2]
```diff
@@ -7,29 +7,22 @@
...
-import com.sequenceiq.freeipa.flow.stack.StackEvent;
-import com.sequenceiq.freeipa.service.stack.StackService;
 
 import reactor.bus.Event;
 import reactor.bus.EventBus;
 
 @Component
 public class FreeIpaFlowManager {
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(FreeIpaFlowManager.class);
```
Comment (topolyai5) [inline freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/flow/FreeIpaFlowManager.java]: You should not remove the LOGGER, you should improve the logging in this class

File: core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/service/TerminationTriggerService.java [hunk 3]
```diff
@@ -58,7 +59,7 @@ private void handleIfStackIsNotTerminated(Stack stack, boolean forced) {
             LOGGER.debug("Found termination flowlog with id [{}] and payload [{}]", flowLog.getFlowId(), flowLog.getPayload());
             handleIfFlowLogExistsForTermination(stack, forced, flowLog);
         } else {
-            LOGGER.warn("Couldn't find FlowLog with 'PRE_TERMINATION_STATE'. Triggering termination");
+            LOGGER.warn("Couldn't find FlowLog with 'INIT_STATE'. Triggering termination");
```
Comment (topolyai5) [inline core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/service/TerminationTriggerService.java]: it is not informative. In the logs, we cannot find easily what flow does not have `INIT_STATE`

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/FreeIpaDeletionService.java [hunk 4]
```diff
@@ -42,9 +61,39 @@ public void delete(String environmentCrn, String accountId) {
...
+        if (optionalFlowLog.isPresent()) {
+            FlowLog flowLog = optionalFlowLog.get();
+            LOGGER.debug("Found termination flowlog with id [{}] and payload [{}]", flowLog.getFlowId(), flowLog.getPayload());
+        } else {
+            fireTerminationEvent(stack);
+        }
+    }
+
+    private void fireTerminationEvent(Stack stack) {
+        LOGGER.warn("Couldn't find FlowLog with 'INIT_STATE'. Triggering termination");
```
Comment (topolyai5) [inline freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/FreeIpaDeletionService.java]: it is not informative. In the logs, we cannot find easily what flow does not have `INIT_STATE`

File: core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/service/TerminationTriggerService.java (hunk 3 above)
Comment (lacikaaa) [reply core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/service/TerminationTriggerService.java]: it's for indicating which path the code took if we are investigating an issue

Comment (lacikaaa) [reply core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/service/TerminationTriggerService.java]: but it should be debug level

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/FreeIpaDeletionService.java (hunk 4 above)
Comment (lacikaaa) [reply freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/FreeIpaDeletionService.java]: modified to debug, it's only to check the flow of the code if there is an issue

File: datalake/src/test/java/com/sequenceiq/datalake/service/sdx/SdxServiceTest.java (hunk 1 above)
Comment (lacikaaa) [reply datalake/src/test/java/com/sequenceiq/datalake/service/sdx/SdxServiceTest.java]: fixed

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/FreeIpaDeletionService.java [hunk 5]
```diff
@@ -42,9 +61,39 @@ public void delete(String environmentCrn, String accountId) {
...
         freeipaJobService.unschedule(stack);
+        if (!stack.isDeleteCompleted()) {
+            handleIfStackIsNotTerminated(stack);
+        } else {
+            LOGGER.debug("Stack is already deleted.");
+        }
+    }
+
+    private void handleIfStackIsNotTerminated(Stack stack) {
+        LOGGER.info("stack {} in environment {} is not deleted.", stack.getName(), stack.getEnvironmentCrn());
```
Comment (keyki) [inline freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/FreeIpaDeletionService.java]: Minor, capital `S`

Comment (lacikaaa) [reply freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/FreeIpaDeletionService.java]: fixed

**Pull request merged**